File: GithubDashboard/app/src/main/java/com/zimpo/githubdashboard/Adapter/RepoAdapter.java
package com.zimpo.githubdashboard.Adapter;

import android.content.Context;
import android.os.Build;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.squareup.picasso.Picasso;
import com.zimpo.githubdashboard.Object.Repo;
import com.zimpo.githubdashboard.R;

import java.util.ArrayList;

import androidx.annotation.RequiresApi;
import androidx.recyclerview.widget.RecyclerView;

public class RepoAdapter extends RecyclerView.Adapter<RepoAdapter.RepoViewHolder> {

    Context context;
    ArrayList<Repo> repos;
    OnItemClickListener listener;

    public interface OnItemClickListener
    {
        void onItemClickListener(Repo repo);
    }

    public RepoAdapter(Context context, OnItemClickListener listener)
    {
        this.context = context;
        this.listener = listener;
        repos = new ArrayList<>();
    }


    public void insertAll(ArrayList<Repo> repos)
    {
        this.repos.clear();
        this.repos.addAll(repos);
        notifyDataSetChanged();
    }

    @Override
    public RepoViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_repo, parent, false);
        RepoViewHolder repoViewHolder = new RepoViewHolder(view);
        return repoViewHolder;
    }

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    @Override
    public void onBindViewHolder(RepoViewHolder holder, int position) {
        holder.Bind(repos.get(position));
    }

    @Override
    public int getItemCount() {
        return repos.size();
    }


    public class RepoViewHolder extends RecyclerView.ViewHolder
    {
        //VARIABLES
        TextView nameTV;

        public RepoViewHolder(View itemView) {
            super(itemView);

            nameTV = (TextView) itemView.findViewById(R.id.nameTV);
        }

        @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
        public void Bind(final Repo repo) {

            nameTV.setText(repo.getName());
        }
    }
}

File: GithubDashboard/app/src/main/java/com/zimpo/githubdashboard/Adapter/RepoAdapter.kt
package com.zimpo.githubdashboard.Adapter

import android.content.Context
import android.os.Build
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView

import com.zimpo.githubdashboard.Model.Repo
import com.zimpo.githubdashboard.R

import java.util.ArrayList

import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.RecyclerView
import com.zimpo.githubdashboard.Api.GithubUserApi
import com.zimpo.githubdashboard.Model.User

class RepoAdapter(private var context: Context, private var listener: OnItemClickListener) : RecyclerView.Adapter<RepoAdapter.RepoViewHolder>() {
    private var repos: ArrayList<Repo>

    interface OnItemClickListener {
        fun onItemClickListener(repo: Repo)

        companion object {
            inline operator fun invoke(crossinline op: (repo: Repo) -> Unit) =
                    object : RepoAdapter.OnItemClickListener {
                        override fun onItemClickListener(repo: Repo) = op(repo)
                    }
        }
    }

    init {
        repos = ArrayList()
    }


    fun insertAll(repos: ArrayList<Repo>) {
        this.repos.clear()
        this.repos.addAll(repos)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RepoViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.view_repo, parent, false)
        return RepoViewHolder(view)
    }

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    override fun onBindViewHolder(holder: RepoViewHolder, position: Int) {
        holder.Bind(repos[position])
    }

    override fun getItemCount(): Int {
        return repos.size
    }


    inner class RepoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        //VARIABLES
        private var repoRL: RelativeLayout
        private var nameTV: TextView

        init {
            repoRL = itemView.findViewById(R.id.repoRL) as RelativeLayout
            nameTV = itemView.findViewById(R.id.nameTV) as TextView
        }

        @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
        fun Bind(repo: Repo) {

            repoRL.setOnClickListener { listener.onItemClickListener(repo) }

            nameTV.setText(repo.name)
        }
    }
}